/1-js/06-data-storage/02-localstorage

perepichai opened the pull request:
Перевод страницы 6-data-storage/02-localstorage/article.md
localStorage и sessionStorage

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/javascript-tutorial/ru.javascript.info?pullRequest=123) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/javascript-tutorial/ru.javascript.info?pullRequest=123) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/javascript-tutorial/ru.javascript.info?pullRequest=123) it.</sub>

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
 # LocalStorage, sessionStorage
 
-Web storage objects `localStorage` and `sessionStorage` allow to save key/value pairs in the browser.
+объекты веб-хранилища `localStorage` и `sessionStorage` позволяют хранить пары key/value в браузере.
```
```suggestion
объекты веб-хранилища `localStorage` и `sessionStorage` позволяют хранить пары ключ/значение в браузере.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+объекты веб-хранилища `localStorage` и `sessionStorage` позволяют хранить пары key/value в браузере.
 
-What's interesting about them is that the data survives a page refresh (for `sessionStorage`) and even a full browser restart (for `localStorage`). We'll see that very soon.
+Что интересно данные "переживают" обновление страницы (для `sessionStorage`) и даже полную перезагрузку браузера (для `localStorage`). Скоро мы это увидим.
```
```suggestion
И что самое интересное: данные сохраняются после обновления страницы (в случае `sessionStorage`) и даже при перезагрузки браузера (при использовании `localStorage`). Скоро мы это увидим.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+Что интересно данные "переживают" обновление страницы (для `sessionStorage`) и даже полную перезагрузку браузера (для `localStorage`). Скоро мы это увидим.
 
-We already have cookies. Why additional objects?
+Но мы также имеем файлы cookie. Зачем тогда эти объекты?
```
```suggestion
Но у нас также есть куки. Зачем тогда эти объекты?
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
-- Unlike cookies, web storage objects are not sent to server with each request. Because of that, we can store much more. Most browsers allow at least 2 megabytes of data (or more) and have settings to configure that.
-- The server can't manipulate storage objects via HTTP headers, everything's done in JavaScript.
-- The storage is bound to the origin (domain/protocol/port triplet). That is, different protocols or subdomains infer different storage objects, they can't access data from each other.
+- В отличие от файлов cookie, объекты веб-хранилища не отправляются на сервер при каждом запросе. Из-за этого мы можем хранить гораздо больше. Большинство браузеров позволяют хранить 2 мегабайта данных (или больше) и имеют параметры для настройки этого.
```
```suggestion
- В отличие от кук, объекты веб-хранилища не отправляются на сервер при каждом запросе. Поэтому мы можем хранить гораздо больше данных. Большинство браузеров могут сохранить как минимум 2 мегабайта данных (или больше), и этот размер можно изменить.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
-- The storage is bound to the origin (domain/protocol/port triplet). That is, different protocols or subdomains infer different storage objects, they can't access data from each other.
+- В отличие от файлов cookie, объекты веб-хранилища не отправляются на сервер при каждом запросе. Из-за этого мы можем хранить гораздо больше. Большинство браузеров позволяют хранить 2 мегабайта данных (или больше) и имеют параметры для настройки этого.
+- Сервер не может манипулировать объектами хранилища через HTTP-заголовки, все делается при помощи JavaScript.
+- Хранилище привязано к источнику (домен/протокол/порт). Это значит, что разные протоколы или субдомены определяют разные объекты хранилища, и они не могут получить доступ к данным друг друга.
```
```suggestion
- Хранилище привязано к источнику (домен/протокол/порт). Это значит, что разные протоколы или поддомены определяют разные объекты хранилища, и они не могут получить доступ к данным друг друга.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
-- Shared between all tabs and windows from the same origin.
-- The data does not expire. It remains after the browser restart and even OS reboot.
+- Доступно на всех вкладках и окнах одного источника.
+- Данные не имеют срока действия. Сохраняются после перезагрузки браузера и даже перезагрузки ОС.
```
```suggestion
- Данные не имеют срока действия. Сохраняются после перезагрузки браузера и даже ОС.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+Это возможно по историческим причинам и как правило работает, но обычно не рекомендуется по двум причинам:
 
-1. If the key is user-generated, it can be anything, like `length` or `toString`, or another built-in method of `localStorage`. In that case `getItem/setItem` work fine, while object-like access fails:
+1. Если ключ генерируется пользователем, то это может быть что угодно в том числе `length` или` toString`, или другой встроенный метод `localStorage`. В этом случае `getItem / setItem` работает нормально, до тех пор, пока объектно-ориентированный доступ не завершится неудачно:
```
```suggestion
1. Если ключ генерируется пользователем, то он может быть каким угодно, включая `length` или `toString`, а также другой встроенный метод `localStorage`. В этом случае `getItem/setItem` работает нормально, до тех пор, пока объектно-ориентированный доступ потерпит неудачу:
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
 ## Looping over keys
 
-Methods provide get/set/remove functionality. But how to get all the keys?
+Методы предоставляют get/set/remove функциональность. Но как получить все ключи?
```
```suggestion
Методы предоставляют для получения, установки и удаления значений. Но как получить все ключи?
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+К сожалению объекты хранилища не итерируются.
 
-One way is to use "array-like" iteration:
+Одним из способов является использование "массиво-подобной" итерации:
```
```suggestion
Одним из способов является использование "массивоподобной" итерации:
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -84,29 +84,29 @@ for(let i=0; i<localStorage.length; i++) {
+Другой способ - использовать объектно-ориентированный `for key in localStorage` цикл.
 
-That iterates over keys, but also outputs few built-in fields that we don't need:
+Здесь перебираются ключи, но вместе с этим выводится несколько встроеных полей которые нам не нужны:
```
```suggestion
Здесь перебираются ключи, но вместе с этим выводится несколько встроенных полей, которые нам не нужны:
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ```
 
-Also it is possible to stringify the whole storage object, e.g. for debugging purposes:
+Также возможно привести к строке весь обьект хранилища, например для задач отладки:
````
```suggestion
Также возможно привести к строке весь объект хранилища, например для задач отладки:
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
-- The data survives page refresh, but not closing/opening the tab.
+- `sessionStorage` существует только в рамках текущей вкладки браузера.
+- Другая вкладка с той же страницей будет иметь другое хранилище.
+- Но оно доступно между фреймами на вкладке (при условии, что они происходят из одного и того же источника).
```
```suggestion
- Но оно доступно между фреймами на вкладке (при условии, что они из одного и того же источника).
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ```
 
-...But if you open the same page in another tab, and try again there, the code above returns `null`, meaning "nothing found".
+...Но если Вы откроете ту же страницу в другой вкладке, и повторите получить данные, код выше вернет `null`, что значит "ничего не найдено".
````
```suggestion
...Но если вы откроете ту же страницу в другой вкладке, и повторите получить данные, код выше вернет `null`, что значит "ничего не найдено".
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
+...Но если Вы откроете ту же страницу в другой вкладке, и повторите получить данные, код выше вернет `null`, что значит "ничего не найдено".
 
-That's exactly because `sessionStorage` is bound not only to the origin, but also to the browser tab. For that reason, `sessionStorage` is used sparingly.
+Это потому, что `sessionStorage` привязан не только к источнику, а и к вкладке браузера. Поэтому, `sessionStorage` используется не часто.
```
```suggestion
Именно из-а того, что `sessionStorage` привязан не только к источнику, а к вкладке браузера. Поэтому `sessionStorage` используется не часто.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ## Storage event
 
-When the data gets updated in `localStorage` or `sessionStorage`, [storage](https://www.w3.org/TR/webstorage/#the-storage-event) event triggers, with properties:
+Когда данные в `localStorage` или` sessionStorage` обновляются, [storage](https://www.w3.org/TR/webstorage/#the-storage-event) срабатывает событие со свойствами:
```
```suggestion
Когда обновляются данные в `localStorage` или `sessionStorage`, генерируется событие [storage](https://www.w3.org/TR/webstorage/#the-storage-event) со следующими свойствами:
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
+- `oldValue` – старые данные (`null` если ключ добавлен впервые).
+- `newValue` – новые данные (`null` если ключ был удален).
+- `url` – url документа, где произошло обновление.
+- `storageArea` – объект `localStorage` или` sessionStorage`, где произошло обновление.
```
```suggestion
- `storageArea` – объект `localStorage` или `sessionStorage`, где произошло обновление.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
-- `storageArea` – either `localStorage` or `sessionStorage` object where the update happened.
+- `key` – ключ который обновился (null если вызвано `.clear()`).
+- `oldValue` – старые данные (`null` если ключ добавлен впервые).
+- `newValue` – новые данные (`null` если ключ был удален).
```
```suggestion
- `newValue` – новые данные (`null`, если ключ был удален).
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
-- `url` – the url of the document where the update happened.
-- `storageArea` – either `localStorage` or `sessionStorage` object where the update happened.
+- `key` – ключ который обновился (null если вызвано `.clear()`).
+- `oldValue` – старые данные (`null` если ключ добавлен впервые).
```
```suggestion
- `oldValue` – старые данные (`null`, если ключ добавлен впервые).
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
-- `newValue` – the new value (`null` if the key is removed).
-- `url` – the url of the document where the update happened.
-- `storageArea` – either `localStorage` or `sessionStorage` object where the update happened.
+- `key` – ключ который обновился (null если вызвано `.clear()`).
```
```suggestion
- `key` – ключ который обновился (`null`, если вызван `.clear()`).
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
+- `storageArea` – объект `localStorage` или` sessionStorage`, где произошло обновление.
 
-The important thing is: the event triggers on all `window` objects where the storage is accessible, except the one that caused it.
+Важно: событие срабатывает на всех `window` обьектах где доступно хранилище, а также там, где оно было вызвано.
```
```suggestion
Важно: событие срабатывает на всех объектах `window`, где доступно хранилище, а также там, где оно было вызвано.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
+Обратите внимание, что событие также содержит: `event.url` - url-адрес документа, в котором данные обновились.
 
-Also, `event.storageArea` contains the storage object -- the event is the same for both `sessionStorage` and `localStorage`, so `storageArea` references the one that was modified. We may event want to set something back in it, to "respond" to a change.
+Также, `event.storageArea` содержит обьект хранилища -- событие одинаковое для `sessionStorage` и `localStorage`, поэтому `storageArea` ссылается на тот который был изменен. Мы можем захотеть что-то вернуть обратно, "ответить" на изменения.
```
```suggestion
Также `event.storageArea` содержит объект хранилища -- событие одинаковое для `sessionStorage` и `localStorage`, поэтому `storageArea` ссылается на тот который был изменен. Мы можем захотеть что-то вернуть обратно, "ответить" на изменения.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-- Contains all the data about the operation, the document `url` and the storage object.
-- Triggers on all `window` objects that have access to the storage except the one that generated it (within a tab for `sessionStorage`, globally for `localStorage`).
+- Срабатывает при вызове `setItem`, `removeItem`, `clear`.
+- Содержит все данные об операции, `url` документа и объект хранилища.
```
```suggestion
- Срабатывает на всех `window` объектах которые имеют доступ к хранилищу, а также там, где оно было сгенерировано (на вкладке для `sessionStorage`, глобально для `localStorage`).
```

perepichai commented:
/done

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-Modern browsers also support [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API), the special API for same-origin inter-window communication, it's more full featured, but less supported. There are libraries that polyfill that API, based on `localStorage`, that make it available everywhere.
+Современные браузеры также поддерживают [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) специальный API для связи между окнами одного источника, он более полнофункциональный, но менее поддерживаемый. Существуют библиотеки (полифилы), которые дополняют API на основе `localStorage`, и делают его доступным везде.
 
 ## Summary
```
```suggestion
## Итого
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-- The data is bound to the origin (domain/port/protocol).
+Объекты веб-хранилища `localStorage` и `sessionStorage` позволяют хранить пары ключ/значение в браузере.
+- `key` и `value` должны быть строками.
+- Лимит 2Мб+, зависит от браузера.
```
```suggestion
- Лимит 2 Мб+, зависит от браузера.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -84,29 +84,29 @@ for(let i=0; i<localStorage.length; i++) {
 // bad try
 for(let key in localStorage) {
-  alert(key); // shows getItem, setItem and other built-in stuff
+  alert(key); // покажет getItem, setItem и другие встроенные вещи
```
```suggestion
  alert(key); // покажет getItem, setItem и другие встроенные свойства
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -84,29 +84,29 @@ for(let i=0; i<localStorage.length; i++) {
 ```
 
-Another way is to use object-specific `for key in localStorage` loop.
+Другой способ - использовать объектно-ориентированный `for key in localStorage` цикл.
````
```suggestion
Другой способ - использовать объектно-ориентированный цикл `for key in localStorage`.
```

lex111 commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+Нам достаточно находится на том же домен/протокол/порт, но при этом путь url может быть разным.
 
-The `localStorage` is shared, so if we set the data in one window, the change becomes visible in the other one.
+`LocalStorage` находится в открытом доступе, поэтому, если мы устанавливаем данные в одном окне, изменения становятся видимыми в другом.
```
```suggestion
Объект `localStorage` находится в открытом доступе, поэтому, если мы устанавливаем данные в одном окне, изменения становятся видимыми в другом.
```

perepichai commented:
/done

iliakan commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ```
 
-Now if both windows are listening for `window.onstorage`, then each one will react on updates that happened in the other one.
+Теперь, если оба окна прослушивать `window.onstorage`, то каждое из них будет реагировать на обновления, произошедшие в другом окне.
````
прослушивать -> слушают

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
+**Это позволяет разным окнам одного источника обмениваться сообщениями.**
 
-Modern browsers also support [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API), the special API for same-origin inter-window communication, it's more full featured, but less supported. There are libraries that polyfill that API, based on `localStorage`, that make it available everywhere.
+Современные браузеры также поддерживают [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) специальный API для связи между окнами одного источника, он более полнофункциональный, но менее поддерживаемый. Существуют библиотеки (полифилы), которые дополняют API на основе `localStorage`, и делают его доступным везде.
```
Существуют библиотеки (полифилы), которые эмулируют это API на основе `localStorage`, и делают его доступным везде.

xcurveballx replied:
перед делают не надо , вроде

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
 |----------------|------------------|
-| Shared between all tabs and windows with the same origin | Visible within a browser tab, including iframes from the same origin |
-| Survives browser restart | Dies on tab close |
+| Совместно используется между всеми вкладками и окнами с одинаковым источником | Отображается на вкладке браузера, включая фреймы из того же источника |
```
| Разделяется в рамках вкладки браузера, среди фреймов из того же источника |

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-- Can use the keys as object properties, in that case `storage` event doesn't trigger.
+- `setItem(key, value)` -- сохранить пару ключ/значение (key/value).
+- `getItem(key)` -- получить данные используя ключ (key).
+- `removeItem(key)` -- удалить пару ключ/значение используя ключ (key).
```
удалить значение по ключу `key`

xcurveballx replied:
, используя

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
+- `getItem(key)` -- получить данные используя ключ (key).
+- `removeItem(key)` -- удалить пару ключ/значение используя ключ (key).
+- `clear()` -- удалить всё.
+- `key(index)` -- получить ключ заданной позиции (index).
```
на заданной позиции

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-- Triggers on all `window` objects that have access to the storage except the one that generated it (within a tab for `sessionStorage`, globally for `localStorage`).
+- Срабатывает при вызове `setItem`, `removeItem`, `clear`.
+- Содержит все данные об операции, `url` документа и объект хранилища.
+- Срабатывает на всех `window` объектах которые имеют доступ к хранилищу, а также там, где оно было сгенерировано (на вкладке для `sessionStorage`, глобально для `localStorage`).
```
Срабатывает на всех объектах `window`, которые

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
-- `key(index)` -- get the key on a given position.
-- `length` -- the number of stored items.
+- `setItem(key, value)` -- сохранить пару ключ/значение (key/value).
+- `getItem(key)` -- получить данные используя ключ (key).
```
получить данные по ключу `key`

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
-- `length` -- the number of stored items.
+- `setItem(key, value)` -- сохранить пару ключ/значение (key/value).
+- `getItem(key)` -- получить данные используя ключ (key).
+- `removeItem(key)` -- удалить пару ключ/значение используя ключ (key).
```
удалить данные с ключом `key`

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+- `clear()` -- удалить всё.
+- `key(index)` -- получить ключ заданной позиции (index).
+- `length` -- количество хранимых пар key/value.
 
```
Как мы видим, интерфейс похож на `Map`, но также запоминается и порядок элементов, можно получить n-й ключ при помощи `.key(n).

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -1,81 +1,81 @@
 ```
 
-We only have to be on the same domain/port/protocol, the url path can be different.
+Нам достаточно находится на том же домен/протокол/порт, но при этом путь url может быть разным.
````
находитЬся

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,81 +1,81 @@
+Это возможно по историческим причинам и как правило работает, но обычно не рекомендуется по двум причинам:
 
-1. If the key is user-generated, it can be anything, like `length` or `toString`, or another built-in method of `localStorage`. In that case `getItem/setItem` work fine, while object-like access fails:
+1. Если ключ генерируется пользователем, то он может быть каким угодно, включая `length` или `toString`, а также другой встроенный метод `localStorage`. В этом случае `getItem/setItem` работает нормально, до тех пор, пока объектно-ориентированный доступ потерпит неудачу:
```
@iliakan может "чтение/запись свойства объекта не пРойдут? а то не могу понять фразу

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -84,29 +84,29 @@ for(let i=0; i<localStorage.length; i++) {
 for(let key in localStorage) {
   if (!localStorage.hasOwnProperty(key)) {
-    continue; // skip keys like "setItem", "getItem" etc
+    continue; // пропустит такие ключи как "setItem", "getItem" и так далее
```
такие, как

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
+Обратите внимание, что ключ и значение должны быть строками.
 
-If we any other type, like a number, or an object, it gets converted to string automatically:
+Если мы используем любой другой тип, например число или объект, то он автоматически преобразуется в строку:
```
например с 2 сторон в ,

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -115,91 +115,91 @@ for(let key of keys) {
-- The data survives page refresh, but not closing/opening the tab.
+- `sessionStorage` существует только в рамках текущей вкладки браузера.
+- Другая вкладка с той же страницей будет иметь другое хранилище.
+- Но оно доступно между фреймами на вкладке (при условии, что они из одного и того же источника).
```
между `<iframe>`

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ```
 
-...Then refresh the page. Now you can still get the data:
+...И обновите страницу. Вы все еще можете получить данные:
````
всЁ ещЁ

iliakan replied:
+1

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -115,91 +115,91 @@ for(let key of keys) {
 ```
 
-...But if you open the same page in another tab, and try again there, the code above returns `null`, meaning "nothing found".
+...Но если вы откроете ту же страницу в другой вкладке, и повторите получить данные, код выше вернет `null`, что значит "ничего не найдено".
````
не надо , перед повторите
и саму фразу я бы написал как "...в другой вкладке и попробуете получить данные снова, то код..."

iliakan replied:
...Но если вы откроете ту же страницу в другой вкладке и повторите получить данные снова, то код выше вернет `null`, что значит "ничего не найдено".

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
+Обратите внимание, что событие также содержит: `event.url` - url-адрес документа, в котором данные обновились.
 
-Also, `event.storageArea` contains the storage object -- the event is the same for both `sessionStorage` and `localStorage`, so `storageArea` references the one that was modified. We may event want to set something back in it, to "respond" to a change.
+Также `event.storageArea` содержит объект хранилища - событие одинаковое для `sessionStorage` и `localStorage`, поэтому `storageArea` ссылается на тот который был изменен. Мы можем захотеть что-то вернуть обратно, "ответить" на изменения.
```
тот, который

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
-- Use `Object.keys` to get all keys.
-- Can use the keys as object properties, in that case `storage` event doesn't trigger.
+- `setItem(key, value)` -- сохранить пару ключ/значение (key/value).
+- `getItem(key)` -- получить данные используя ключ (key).
```
, используя

xcurveballx commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -208,40 +208,40 @@ window.onstorage = event => {
+- Можно использовать ключи в качестве свойств объекта, в этом случае событие `storage` не страбатывает.
 
-Storage event:
+Событие хранилища:
```
Событие storage

xcurveballx commented:
пинганем @perepichai

xcurveballx commented:
@perepichai там еще конфликт есть, вероятно были изменения в английской версии не так давно. Пожалуйста, решите его тоже.

perepichai commented:
/done

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,83 +1,83 @@
+Объекты веб-хранилища `localStorage` и `sessionStorage` позволяют хранить пары ключ/значение в браузере.
 
-What's interesting about them is that the data survives a page refresh (for `sessionStorage`) and even a full browser restart (for `localStorage`). We'll see that very soon.
+Что в них важно - данные, которые в них записаны, сохраняются после обновления страницы (в случае sessionStorage) и даже после перезагрузки браузера (при использовании localStorage). Скоро мы это увидим.
```
Что важно - ...

iliakan commented on 6-data-storage/02-localstorage/article.md:
````diff
@@ -1,83 +1,83 @@
 ```
 
-We only have to be on the same domain/port/protocol, the url path can be different.
+Нам достаточно находится на том же источнике (домен/протокол/порт), при этом URL-путь может быть разным.
````
Нам достаточно находиться на том же источнике (то же  домен/протокол/порт), при этом URL-путь может быть разным.

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,83 +1,83 @@
+Нам достаточно находится на том же источнике (домен/протокол/порт), при этом URL-путь может быть разным.
 
-The `localStorage` is shared, so if we set the data in one window, the change becomes visible in the other one.
+Объект `localStorage` разделяется, поэтому, если мы устанавливаем данные в одном окне, изменения становятся видимыми в другом.
```
Объект `localStorage` разделяется между всеми окнами с одного источника, поэтому, если мы устанавливаем данные в одном окне, изменения становятся видимыми в другом.

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -1,83 +1,83 @@
+## Перебор ключей
 
-As we've seen, the methods provide get/set/remove functionality. But how to get all saved values or keys?
+Методы, которые мы видим, позволяют читать/писать/удалять данные. А как получить все значения или ключи?
```
Методы, которые мы видим, позволяют читать/писать/удалять данные по ключу. А как получить все значения или ключи?

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -117,91 +117,91 @@ for(let key of keys) {
-- The data survives page refresh, but not closing/opening the tab.
+- `sessionStorage` существует только в рамках текущей вкладки браузера.
+- Другая вкладка с той же страницей будет иметь другое хранилище.
+- Но оно доступно между фреймами на вкладке (при условии, что они из одного и того же источника).
```
доступно -> разделяется

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -117,91 +117,91 @@ for(let key of keys) {
+...Но если вы откроете ту же страницу в другой вкладке, и повторите получить данные, код выше вернет `null`, что значит "ничего не найдено".
 
-That's exactly because `sessionStorage` is bound not only to the origin, but also to the browser tab. For that reason, `sessionStorage` is used sparingly.
+Именно из-за того, что `sessionStorage` привязан не только к источнику, а к вкладке браузера. Поэтому `sessionStorage` используется нечасто.
```
, а к вкладке браузера -> , но также и к вкладке браузера

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -117,91 +117,91 @@ for(let key of keys) {
+Именно из-за того, что `sessionStorage` привязан не только к источнику, а к вкладке браузера. Поэтому `sessionStorage` используется нечасто.
 
-## Storage event
+## Событие `storage`
```
Событие "storage"

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -117,91 +117,91 @@ for(let key of keys) {
+Давайте уточним.
 
-Imagine, you have two windows with the same site in each. So `localStorage` is shared between them.
+Представьте, что у вас есть два окна с одним и тем же сайтом. `localStorage` разделяется между ними.
```
Представьте, что у вас есть два окна с одним и тем же сайтом. Хранилище `localStorage` разделяется между ними.

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -210,40 +210,40 @@ window.onstorage = event => {
+**Это позволяет разным окнам одного источника обмениваться сообщениями.**
 
-Modern browsers also support [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API), the special API for same-origin inter-window communication, it's more full featured, but less supported. There are libraries that polyfill that API, based on `localStorage`, that make it available everywhere.
+Современные браузеры также поддерживают [Broadcast channel API](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) специальный API для связи между окнами одного источника, он более полнофункциональный, но менее поддерживаемый. Существуют библиотеки (полифилы), которые эмулируют это API на основе `localStorage`, и делают его доступным везде.
```
связи между окнами -> рассылки сообщений по всем окнам

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -210,40 +210,40 @@ window.onstorage = event => {
-| Shared between all tabs and windows with the same origin | Visible within a browser tab, including iframes from the same origin |
-| Survives browser restart | Dies on tab close |
+| Совместно используется между всеми вкладками и окнами с одинаковым источником | Разделяется в рамках вкладки браузера, среди фреймов из того же источника |
+| "Переживает" перезапуск браузера | "Умирает" при закрытии вкладки |
```
| "Переживает" перезапуск браузера | "Переживает" перезагрузку страницы (но не закрытие вкладки) |

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -210,40 +210,40 @@ window.onstorage = event => {
+- `getItem(key)` -- получить данные по ключу `key`.
+- `removeItem(key)` -- удалить значение по ключу `key`.
+- `clear()` -- удалить всё.
+- `key(index)` -- получить ключ на заданной позиции (index).
```
- `key(index)` -- получить ключ номер `index`.

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -210,40 +210,40 @@ window.onstorage = event => {
+- `key(index)` -- получить ключ на заданной позиции (index).
+- `length` -- количество хранимых пар key/value.
+- Используйте `Object.keys` для получения всех ключей.
+- Можно использовать ключи в качестве свойств объекта, в этом случае событие `storage` не срабатывает.
```
- Можно обращаться к ключам как к свойствам объекта, в этом случае событие `storage` не срабатывает.

iliakan commented on 6-data-storage/02-localstorage/article.md:
```diff
@@ -210,40 +210,40 @@ window.onstorage = event => {
+- Можно использовать ключи в качестве свойств объекта, в этом случае событие `storage` не срабатывает.
 
-Storage event:
+Событие хранилища:
```
Событие `storage`:

perepichai commented:
Как просмотреть все запрашиваемые изменения, нужно пройтись по всем сообщениям здесь в переписке?

perepichai commented:
может подскажете в чем дело, не могу сделать push. Ругается на это:
 ! [rejected]        translate -> translate (non-fast-forward)
error: failed to push some refs to 'https://github.com/perepichai/ru.javascript.info.git'
hint: Updates were rejected because the tip of your current branch is behind

lex111 commented:
А если сделать сначала pull, а потом push, то же будет ошибка?

perepichai commented:
> 
> 
> А если сделать сначала pull, а потом push, то же будет ошибка?

Нужно выполнить git pull origin master  ?

perepichai commented:
> 
> 
> > А если сделать сначала pull, а потом push, то же будет ошибка?
> 
> Нужно выполнить git pull origin master ?
Или git pull origin translate ?

lex111 commented:
> git pull origin translate

perepichai commented:
> 
> 
> > git pull origin translate

From https://github.com/perepichai/ru.javascript.info
 * branch            translate  -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
        figures.sketch
Please, commit your changes or stash them before you can merge.
Aborting

lex111 commented:
Откатите изменение файла figures.sketch - `git checkout figures.sketch`, затем снова pull.

perepichai commented:
> 
> 
> Откатите изменение файла figures.sketch - `git checkout figures.sketch`, затем снова pull.

Сделал, все равно ругается на тот же файл figures.sketch
 * branch            translate  -> FETCH_HEAD
error: Your local changes to the following files would be overwritten by merge:
        figures.sketch
Please, commit your changes or stash them before you can merge.
Aborting

perepichai commented:
> 
> 
> > Откатите изменение файла figures.sketch - `git checkout figures.sketch`, затем снова pull.
> 
> Сделал, все равно ругается на тот же файл figures.sketch
> 
>     * branch            translate  -> FETCH_HEAD
>       error: Your local changes to the following files would be overwritten by merge:
>       figures.sketch
>       Please, commit your changes or stash them before you can merge.
>       Aborting

Может можно ещё что то предпринять?

lex111 commented:
А если сделать `git reset --hard`?

iliakan commented:
git checkout HEAD figures.sketch

perepichai commented:
> 
> 
> git checkout HEAD figures.sketch (вы, видимо, проиндексировали файл, ваша команда берет файл из индекса, а репо-версия в HEAD)

сделал git reset --hard
потом pull
и после делаю push - выдает ошибку:
 ! [rejected]        translate -> translate (non-fast-forward)
error: failed to push some refs to 'https://github.com/perepichai/ru.javascript.info.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

xcurveballx commented:
@perepichai  так а вы решили конфликт после того, как спулили?

perepichai commented:
> 
> 
> @perepichai так а вы решили конфликт и закоммитили изменения после того, как спулили?

Нет, немного не понятно было в чем дело. Попробую исправить конфликты

perepichai commented:
Конфликты исправил. Сделал коммит. Вечером еще внесу изменения

perepichai commented:
уже запутался с этими конфликтами, но возможно все исправил как нужно

perepichai commented:
/done

iliakan commented:
куча каких-то новых файлов с ~HEAD, возможно, по ошибке добавили?

xcurveballx commented:
у меня при клонировании этого репо отключается интернет)) @perepichai попало много лишних файлов. Получите список коммитов в той ветке, чекаутите каждый по очереди в обратном порядке, пока не исчезнут лишние файлы, а потом сделайте хард ресет и форс пуш.

xcurveballx commented:
как вариант, можно обновить форк и сделать новый PR, скопировав туда перевод главы.

perepichai commented:
А вот это мне нравиться больше) в какой то момент все пошло не так. А то вместо перевода борюсь с гитом

iliakan commented:
Ждём =)

perepichai commented:
/done

iliakan commented:
В Files changed по-прежнему много файлов

iliakan commented:
И еще правки не учтены.

perepichai commented:
> 
> 
> Правки не учтены, файлы лишние.

Как можно от этих файлов избавиться? Склонировал форк в новую папку. Затем сделал git pull с основного репозитория в мастер. git pull upstream master. Далее git merge master. Затем сделал коммит только одного файла и запушил. Новый pull request мне поможет?

iliakan commented:
Реально ты менял только текст 1го файла, да?

iliakan commented:
Я там видел - мои правки по тексту не все учтены, как минимум одна =)

perepichai commented:
> 
> 
> Реально ты менял только текст 1го файла, да?

меняю только один файл и добавляю его в коммит. Не могу понять каким образом добавились еще 45 файлов.

perepichai commented:
> 
> 
> Я там видел - мои правки по тексту не все учтены, как минимум одна =)

Пересмотрю)

iliakan commented:
Прими все правки по этому файлу и создай абсолютно новый PR только с ним. Можешь репу с нуля переклонировать или как проще =)

perepichai commented:
> 
> 
> Прими все правки по этому файлу и создай абсолютно новый PR только с ним. Можешь репу с нуля переклонировать или как проще =)

Сейчас вот такая картина (если сделаю PR должно быть ок?):
PS C:\development\javascriptru\new\ru.javascript.info> git status
On branch translate
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   6-data-storage/02-localstorage/article.md

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   1-js/09-classes/02-class-inheritance/3-class-extend-object/rabbit-extends-object.png
        modified:   1-js/09-classes/02-class-inheritance/3-class-extend-object/rabbit-extends-object@2x.png
        modified:   1-js/09-classes/02-class-inheritance/animal-rabbit-extends.png
        modified:   1-js/09-classes/02-class-inheritance/animal-rabbit-extends@2x.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-array-object.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-array-object@2x.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-animal-2.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-animal-2@2x.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-animal.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-animal@2x.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-run-animal.png
        modified:   1-js/09-classes/02-class-inheritance/class-inheritance-rabbit-run-animal@2x.png
        modified:   1-js/09-classes/02-class-inheritance/rabbit-animal-independent-animal.png
        modified:   1-js/09-classes/02-class-inheritance/rabbit-animal-independent-animal@2x.png
        modified:   1-js/09-classes/02-class-inheritance/rabbit-animal-independent-rabbit.png
        modified:   1-js/09-classes/02-class-inheritance/rabbit-animal-independent-rabbit@2x.png
        modified:   1-js/09-classes/02-class-inheritance/super-homeobject-wrong.png
        modified:   1-js/09-classes/02-class-inheritance/super-homeobject-wrong@2x.png
        modified:   1-js/09-classes/02-class-inheritance/this-super-loop.png
        modified:   1-js/09-classes/02-class-inheritance/this-super-loop@2x.png
        modified:   1-js/09-classes/03-static-properties-methods/animal-rabbit-static.png
        modified:   1-js/09-classes/03-static-properties-methods/animal-rabbit-static@2x.png
        modified:   1-js/09-classes/04-private-protected-properties-methods/coffee-inside.jpg
        modified:   1-js/09-classes/04-private-protected-properties-methods/coffee.jpg
        modified:   1-js/09-classes/05-extend-natives/object-date-inheritance.png
        modified:   1-js/09-classes/05-extend-natives/object-date-inheritance@2x.png
        modified:   1-js/09-classes/06-instanceof/instanceof.png
        modified:   1-js/09-classes/06-instanceof/instanceof@2x.png
        modified:   1-js/09-classes/07-mixins/mixin-inheritance.png
        modified:   1-js/09-classes/07-mixins/mixin-inheritance@2x.png
        modified:   10-misc/01-proxy/proxy-inherit.png
        modified:   10-misc/01-proxy/proxy-inherit@2x.png
        modified:   10-misc/01-proxy/proxy.png
        modified:   10-misc/01-proxy/proxy@2x.png
        modified:   2-ui/3-event-details/5-keyboard-events/german-layout.png
        modified:   2-ui/3-event-details/5-keyboard-events/german-layout@2x.png
        modified:   2-ui/3-event-details/5-keyboard-events/us-layout.png
        modified:   2-ui/3-event-details/5-keyboard-events/us-layout@2x.png

xcurveballx commented:
Там теперь 61 файл, а было 45)) Может просто создать новый форк и новый PR и скопировать только статью с переводом?

iliakan commented:
да

perepichai commented:
> 
> 
> да

Сделал новый форк

xcurveballx commented:
В связи с перепетиями с гитом этот PR зашел в тупик, продолжение в #301
